core/src/test/java/org/infinispan/stream/BaseStreamTest.java

```diff
@@ -255,6 +254,100 @@ public void testObjForEach() {
       }
    }
 
+   private static class ForEachInjected<E> implements Consumer<E>,
+           CacheAware<Integer, String>, Serializable {
+      private transient Cache<?, ?> cache;
+      private final int cacheOffset;
+      private final int atomicOffset;
+
+      private final ToIntFunction<? super E> function;
+
+      private ForEachInjected(int cacheOffset, int atomicOffset, ToIntFunction<? super E> function) {
+         this.cacheOffset = cacheOffset;
+         this.atomicOffset = atomicOffset;
+         this.function = function;
+      }
+
+      @Override
+      public void injectCache(Cache<Integer, String> cache) {
+         this.cache = cache;
+      }
+
+      @Override
+      public void accept(E entry) {
+         Cache<?, ?> cache = getForEachObject(cacheOffset);
+         if (cache != null && this.cache != null && cache.getName().equals(this.cache.getName())) {
```

Comment:
We could do that.  I am not sure how the exception will propagate, but I guess we can find out :)